src/test/kotlin/nomic/unit/config/security/filters/JWTAuthenticationFilterTest.kt

```diff
@@ -0,0 +1,110 @@
+    fun init() {
+        // Hacky solution to ensure that the security context doesn't persist between tests
+        SecurityContextHolder.clearContext()
+    }
+
+    @Test
+    fun filter_validToken_succeeds() {
+        // Since we mock the token registry, we don't need a legit token.
+        val token1 = "LegitToken1"
+        val user1 = User(100, "Frodo Baggins")
+
+        val token2 = "LegitToken2"
+        val user2 = User(12, "Aragorn")
+
+        val tokenRegistry = mock<ITokenRegistry> {
+            on { validateToken(token1) } doReturn TokenValidationResult(true, user1)
+            on { validateToken(token2) } doReturn TokenValidationResult(true, user2)
+        }
+
+        val mockMvc = MockMvcBuilders
+            .standaloneSetup(TestController())
+            .addFilter<StandaloneMockMvcBuilder>(JWTAuthenticationSecurityFilter(tokenRegistry))
+            .build()
+
+        val request1 = MockMvcRequestBuilders.get("/test")
+            .header(HttpHeaders.AUTHORIZATION, "Bearer $token1")
+
+        mockMvc.perform(request1)
+            .andExpect(content().string(user1.id.toString()))
+            .andExpect(status().isOk)
+
+        val request2 = MockMvcRequestBuilders.get("/test")
+            .header(HttpHeaders.AUTHORIZATION, "Bearer $token2")
+
+        mockMvc.perform(request2)
+            .andExpect(status().isOk)
+            .andExpect(content().string(user2.id.toString()))
+    }
+
+    @Test
+    fun filter_invalidToken_fails() {
+        // Since we mock the token registry, we don't need a legit token.
+        val token = "BadToken"
+
+        val tokenRegistry = mock<ITokenRegistry> {
+            on { validateToken(token) } doReturn TokenValidationResult(false)
+        }
+
+        val mockMvc = MockMvcBuilders
+            .standaloneSetup(TestController())
+            .addFilter<StandaloneMockMvcBuilder>(JWTAuthenticationSecurityFilter(tokenRegistry))
+            .build()
+
+        val request = MockMvcRequestBuilders.get("/test")
+            .header(HttpHeaders.AUTHORIZATION, "Bearer $token")
+
+        mockMvc.perform(request)
+            .andExpect(status().isOk)
+            .andExpect(content().string("null"))
+    }
+
+    @Test
+    fun filter_noToken_fails() {
+        val tokenRegistry = mock<ITokenRegistry> { }
+
+        val mockMvc = MockMvcBuilders
+            .standaloneSetup(TestController())
+            .addFilter<StandaloneMockMvcBuilder>(JWTAuthenticationSecurityFilter(tokenRegistry))
+            .build()
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/test"))
+            .andExpect(status().isOk)
+            .andExpect(content().string("null"))
+    }
+
+    @RestController
+    private class TestController {
+
+        @GetMapping("/test")
+        fun test(): String {
+            val context = SecurityContextHolder.getContext()
+            val user = context.authentication?.principal as User?
+            return user?.id.toString()
+        }
+    }
```

Comment:
There should be a comment explaining why the mock is here and what purpose it is serving to the test